examples/step-68/step-68.cc

```diff
@@ -0,0 +1,810 @@
+  };
+
+  // There remains the task of declaring what run-time parameters we can accept
+  // in input files. Since we have a very limited number of parameters, all
+  // parameters are declared in the same section.
+  ParticleTrackingParameters::ParticleTrackingParameters()
+    : ParameterAcceptor("Particle Tracking Problem/")
+  {
+    add_parameter(
+      "Velocity degree", velocity_degree, "", this->prm, Patterns::Integer(1));
+
+    add_parameter("Output frequency",
+                  output_frequency,
+                  "Iteration frequency at which output results are written");
+
+    add_parameter("Repartition frequency",
+                  repartition_frequency,
+                  "Iteration frequency at which the mesh is load balanced");
+
+    add_parameter("Output directory", output_directory);
+
+    add_parameter("Time step", time_step);
+
+    add_parameter("Final time", final_time, "End time of the simulation");
+
+    add_parameter("Fluid refinement",
+                  fluid_refinement,
+                  "Refinement level of the fluid domain");
+
+    add_parameter(
+      "Particle insertion refinement",
+      particle_insertion_refinement,
+      "Refinement of the volumetric mesh used to insert the particles");
+  }
+
+  // @sect3{Velocity profile}
+
+  // The velocity profile is provided as a Function object.
+  // This function is hard-coded within
+  // the example.
+  template <int dim>
+  class Vortex : public Function<dim>
+  {
+  public:
+    Vortex()
+      : Function<dim>(dim)
+    {}
+    virtual void vector_value(const Point<dim> &point,
+                              Vector<double> &  values) const override;
+  };
+
+  template <int dim>
+  void Vortex<dim>::vector_value(const Point<dim> &point,
+                                 Vector<double> &  values) const
+  {
+    const double T = 4;
+    // The velocity profile for the Rayleigh-Kothe vertex is time-dependent.
+    // Consequently, the current time in the
+    // simulation (t) must be gathered from the Function object.
+    const double t = this->get_time();
+
+    const double px = numbers::PI * point(0);
+    const double py = numbers::PI * point(1);
+    const double pt = numbers::PI / T * t;
+
+    values[0] = -2 * cos(pt) * pow(sin(px), 2) * sin(py) * cos(py);
+    values[1] = 2 * cos(pt) * pow(sin(py), 2) * sin(px) * cos(px);
+    if (dim == 3)
+      {
+        values[2] = 0;
+      }
+  }
+
+  // @sect3{The <code>ParticleTracking</code> class declaration}
+
+  // We are now ready to introduce the main class of our tutorial program.
+  template <int dim>
+  class ParticleTracking
+  {
+  public:
+    ParticleTracking(const ParticleTrackingParameters &par,
+                     const bool                        interpolated_velocity);
+    void run();
+
+  private:
+    // The particles_generation function is responsible for the initial
+    // generation of the particles on top of the background grid
+    void particles_generation();
```

Comment:
That's a slightly funny name for a function. Function names should have a verb in it to indicate an *action*. May I suggest `generate_particles()`?